controllers/envoy/envoyconfig_controller.go

```diff
@@ -97,14 +82,14 @@ func (r *EnvoyConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 	}
 
 	// Update the ConfigRevisions list in the status
-	if err := r.consolidateRevisionList(ctx, ec, desiredVersion); err != nil {
+	if err := r.reconcileRevisionList(ctx, ec, desiredVersion); err != nil {
```

Comment:
The change of name for this function is because so far envoyconfigrevisions were simply appended to the end of the list. Now the list is fully regenerated on each reconcile based on a specific criteria. This was required in order to recreate the full list when moving from v2 to v3 and viceversa.